openrpc.json

```diff
@@ -283,6 +283,68 @@
         }
       ]
     },
+    {
+      "name": "wallet_revokePermissions",
+      "tags": [
+        {
+          "$ref": "#/components/tags/MetaMask"
+        }
+      ],
+      "summary": "Revoke wallet permissions.",
+      "description": "Revoke previously granted permissions for a specific dapp identified by origin (invoker)",
+      "paramStructure": "by-name",
+      "params": [
+        {
+          "name": "permission",
+          "required": false,
+          "schema": {
+            "type": "object",
+            "properties": {
+              "target": {
+                "type": "string",
+                "description": "The target name of the permission",
+                "enum": ["eth_accounts"]
+              },
+              "caveatType": {
+                "type": "string",
+                "description": "The type of the caveat to update",
+                "enum": ["restrictReturnedAccounts"]
+              },
+              "caveatValue": {
+                "type": "array",
+                "description": "The value of the caveat to be revoked from the target permission",
+                "items": {
+                  "type": "string"
+                }
+              }
+            }
+          }
+        }
+      ],
+      "result": {
+        "name": "UpdatedPermissionsAfterRevoke",
+        "description": "The list of the revoked permissions objects",
+        "schema": {
+          "$ref": "#/components/schemas/PermissionsList"
+        }
+      },
+      "errors": [
+        {
+          "code": 404,
```

Comment:
This should be a JSON-RPC error code. Consult with @shanejonas / @BelfordZ.

Reply:
you can check out https://github.com/MetaMask/rpc-errors/blob/main/src/error-constants.ts for our current error codes.

This looks like maybe `-32002` - resource unavailable